src/main/java/zosjobs/CancelJobs.java

```diff
@@ -0,0 +1,70 @@
+package zosjobs;
+
+import core.ZOSConnection;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import rest.*;
+import utility.Util;
+import utility.UtilIO;
+import zosjobs.input.CancelJobParams;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class CancelJobs {
+
+    private static final Logger LOG = LogManager.getLogger(CancelJobs.class);
+    private final ZOSConnection connection;
+
+    /**
+     * CancelJobs Constructor
+     *
+     * @param connection connection information, see ZOSConnection object
+     * @author Nikunj Goyal
+     */
+    public CancelJobs(ZOSConnection connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * Cancel a job that resides in a z/OS data set.
+     *
+     * @return job document with details about the submitted job
+     * @throws Exception error on submitting
+     * @author Frank Giordano
+     */
+    public Response cancelJob() throws Exception {
+        return this.cancelJobsCommon(null);
+    }
+
+
+    public Response cancelJobsCommon(CancelJobParams parms) throws Exception {
+        Util.checkNullParameter(parms == null, "parms is null");
+        Util.checkStateParameter(!parms.getJobId().isPresent(), "jobid not specified");
+        Util.checkStateParameter(!parms.getJobname().isPresent(), "jobname not specified");
```

Comment:
the project is setup for java 11. with java 11 you can use the cleaner version of !parms.getVersion().isPresent() which is parms.getJobname().isEmpty() instead etc..